Makefile.toml

```diff
@@ -0,0 +1,444 @@
+    "test-wasm-example-app",
+    "test-wasm-feature",
+]
+[tasks.test-example]
+category = "Tests"
+dependencies = [
+    "test-cpp-example",
+    "test-c-example",
+    "test-c2-example",
+    "test-wasm-example",
+]
+
+[tasks.test-feature]
+category = "Tests"
+dependencies = [
+    "test-cpp-feature",
+    "test-wasm-feature",
+]
+
+[tasks.test-cpp-example]
+category = "Tests"
+dependencies = ["build-example"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd example/cpp
+rm tests/*.out
+exec make
+'''
+
+[tasks.test-cpp-feature]
+category = "Tests"
+dependencies = ["build-feature"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd feature_tests/cpp
+rm tests/*.out
+exec make
+'''
+
+[tasks.test-c-example]
+category = "Tests"
+dependencies = ["build-example"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd example/c
+rm *.out
+exec make
+'''
+
+[tasks.test-c2-example]
+category = "Tests"
+dependencies = ["build-example"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd example/c
+rm *.out
+exec make
+'''
+
+[tasks.test-wasm-example]
+category = "Tests"
+dependencies = ["build-example-wasm"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd example/js/
+cp ../../target/wasm32-unknown-unknown/debug/diplomat_example.wasm lib/api/diplomat_example.wasm
+cd lib
+exec npm install
+exec npm run test
+'''
+
+[tasks.test-wasm-example-app]
+category = "Tests"
+dependencies = ["build-example-wasm"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd example/js/
+cp ../../target/wasm32-unknown-unknown/debug/diplomat_example.wasm lib/api/diplomat_example.wasm
+cd app
+exec npm install
+exec npm run build
+'''
+
+[tasks.test-wasm-feature]
+category = "Tests"
+dependencies = ["build-feature-wasm"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd feature_tests/js/
+cp ../../target/wasm32-unknown-unknown/debug/diplomat_feature_tests.wasm api/diplomat_feature_tests.wasm
+cd app
+exec npm install
+exec npm run test
+'''
+
+[tasks.test-c2-example-self-contained]
+category = "Tests"
+dependencies = ["build-example"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd example/c2/include
+files_list = glob_array "./*.h"
+for file in ${files_list}
+    exec gcc ${file}
+end
+'''
+
+[tasks.test-c2-feature-self-contained]
+category = "Tests"
+dependencies = ["build-example"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+cd feature_tests/c2/include
+files_list = glob_array "./*.h"
+for file in ${files_list}
+    exec gcc ${file}
+end
+'''
+
+# Generation
+
+# Generic task for generating code
+# Usage: cargo make gen-generic feature_tests <backend name> <output folder name> <optional docs folder> <optional additional args>
+[tasks.gen-generic]
+category = "Plumbing"
+dependencies = ["build-tool"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+path = array_get ${@} 0
```

Comment:
Praise: Glad you got Makefile arguments working

Reply:
it's an absolute mess lol, but it works

Reply:
https://github.com/rust-diplomat/diplomat/pull/304